prow/cmd/deck/main.go

```diff
@@ -157,19 +158,34 @@ var (
 			},
 			[]string{"path", "method", "status"},
 		),
+		httpRequestSize: prometheus.NewHistogramVec(
+			prometheus.HistogramOpts{
+				Name:    "deck_http_request_size_bytes",
```

Comment:
Why the `deck_` prefix here? The metric name would follow prometheus best practices more closely without it.

Reply:
> deck_

It indicates where the data come without looking at the label.
[best practice doc](https://prometheus.io/docs/practices/naming/) has `prometheus_notifications_total` as an example.
It looks to me `prometheus_` has similar meaning.

I am no expert of prometheus.
Would it cause performance issue (eg query answering) if all services share the same metrics name like:

```
http_request_size_bytes{job="deck"}
http_request_size_bytes{job="sinker"}
...
```

Reply:
> I am no expert of prometheus.
Would it cause performance issue (eg query answering) if all services share the same metrics name like:
>```
>http_request_size_bytes{job="deck"}
>http_request_size_bytes{job="sinker"}
>...

This is the preferred naming in prometheus... this let's you easily aggregate `http_request_size_bytes` across all your services and drill down on a service using labels. Without this, if you want to compare `http_request_size_bytes` across two services, you would need to do two queries (or maybe a join?). Label cardinality is more of a problem with _lots_ of different label keys, it's pretty easy to have lots of different label values and shouldn't be an issue.

Reply:
> It indicates where the data come without looking at the label.
best practice doc has prometheus_notifications_total as an example.
>It looks to me prometheus_ has similar meaning.

Yeah, this can be a little confusing in the docs... in this case, the prefix they are mentioning is `http_`. If this were a grpc service, we might use `grpc_` as the prefix. But you really shouldn't have two prefixes, so no `prometheus_http_` or `prometheus_http_request_size_bytes`.

Reply:
Thanks for your replies.